src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/InitializationService.java

```diff
@@ -0,0 +1,315 @@
+        ownerClient.setConf(serverConf);
+        // again for partial init support: if the client already exists there is no reason to add it again
+        if (!serverConf.getClient().contains(ownerClient)) {
+            serverConf.getClient().add(ownerClient);
+        }
+        serverConf.setOwner(ownerClient);
+        serverConf.setServerCode(securityServerCode);
+        return serverConf;
+    }
+
+    /**
+     * Helper to check for warnings
+     * @param ownerClientId
+     * @param securityServerCode
+     * @throws UnhandledWarningsException
+     */
+    private void checkForWarnings(ClientId ownerClientId, String securityServerCode)
+            throws UnhandledWarningsException {
+        String ownerMemberName = globalConfFacade.getMemberName(ownerClientId);
+        SecurityServerId serverId = SecurityServerId.create(ownerClientId, securityServerCode);
+        List<WarningDeviation> warnings = new ArrayList<>();
+        if (StringUtils.isEmpty(ownerMemberName)) {
+            WarningDeviation memberWarning = new WarningDeviation(WARNING_INIT_UNREGISTERED_MEMBER,
+                    ownerClientId.toShortString());
+            warnings.add(memberWarning);
+        }
+        if (globalConfFacade.existsSecurityServer(serverId)) {
+            WarningDeviation memberWarning = new WarningDeviation(WARNING_INIT_SERVER_ID_EXISTS,
+                    serverId.toShortString());
+            warnings.add(memberWarning);
+        }
+        if (!warnings.isEmpty()) {
+            throw new UnhandledWarningsException(warnings);
+        }
+    }
+
+    /**
+     * Helper to create an initial client
+     * @param clientId
+     * @return
+     */
+    private ClientType getInitialClient(ClientId clientId) {
+        ClientType localClient = clientService.getLocalClient(clientId);
+        if (localClient == null) {
+            localClient = new ClientType();
+            localClient.setIdentifier(clientId);
+            localClient.setClientStatus(ClientType.STATUS_SAVED);
+            localClient.setIsAuthentication(IsAuthentication.SSLAUTH.name());
+        }
+        return localClient;
+    }
+
+    /**
+     * Verify that the initialization process can proceed and that the security server has not already been
+     * initialized. This means verifying that an anchor has been imported,
+     * server conf does not exist and a software token has not yet been initialized. This method could also be
+     * changed into throwing an UnhandledWarningsException with warning metadata instead of the current
+     * InitializationException if partial init needs to be supported in the future
+     * @throws InitializationException if server conf exists OR software token is already initialized
+     */
+    private void verifyInitializationPrerequisites() throws InitializationException {
+        boolean isServerConfInitialized = serverConfService.isServerConfInitialized();
```

Comment:
Based on the discussion, it seems that partial init should be implemented

Reply:
Partial init has been implemented. 

Primary changes:

The init endpoint itself does not have any mandatory parameters anymore because the user must be able to initialize for example only the software token (while skipping other init parts such as server code and server owner). However the parameters are required for each missing part of the initialization. For example: if server owner has not been initialized, the server owner parameters are mandatory in the request, otherwise a `MissingInitParamsException` is thrown.

Also prerequisite check now throws warnings (depending on `ignore_warnings` flag) if the Security Server already has some initialized parts.